Yatzy/Category.cs

```diff
@@ -106,76 +134,54 @@ public int LargeStraight()
         public int FullHouse()
         {
             int totalScore = 0;
-            int threeOfAKindPoints = ThreeOfAKind();
-            if (threeOfAKindPoints == 0)
-            {
-                return totalScore;
-            }
-
-            int threeOfAKindNumber = GetNumbersWithMultipleOccurence(3).First();
-            _diceNumbers.RemoveAll(number => number == threeOfAKindNumber);
-            int pairPoints = Pair();
-            if (pairPoints == 0)
+            var threeOfAKind = _diceNumbersWithCount.FirstOrDefault(diceCount => diceCount.Value == 3);
+            if (threeOfAKind.Key > 0)
             {
-                return totalScore;
+                var pair = _diceNumbersWithCount.FirstOrDefault(diceCount => diceCount.Value == 2);
+                if (pair.Key > 0)
+                {
+                    totalScore = (threeOfAKind.Key * 3) + (pair.Key * 2);
+                }
```

Comment:
maybe can move `var pair ....` line just after `var threeOfAKind ...` line
Then `return (threeOfAKind.Key > 0) && (pair.Key > 0) ? (threeOfAKind.Key * 3) + (pair.Key * 2) : 0;`
All if statements can then be removed.